src/components/typography/text/text.js

```diff
@@ -33,14 +36,58 @@ const Headline = props => {
 
 Headline.displayName = 'TextHeadline';
 Headline.propTypes = {
-  elementType: PropTypes.oneOf(['h1', 'h2', 'h3']).isRequired,
+  as(props, propName, componentName, ...rest) {
+    // is required if the prop 'elementType' is not defined
+    if (!props.elementType) {
+      return PropTypes.oneOf(['h1', 'h2', 'h3']).isRequired(
+        props,
+        propName,
+        componentName,
+        ...rest
+      );
+    }
+    return PropTypes.oneOf(['h1', 'h2', 'h3'])(
+      props,
+      propName,
+      componentName,
+      ...rest
+    );
+  },
+  elementType(props, propName, componentName, ...rest) {
+    if (props[propName] != null) {
+      const message = oneLine`
+            "${propName}" property of "${componentName}" has been deprecated
+            and will be removed in the next major version.\n
+            Please use "as" prop instead.`;
+      warning(false, message);
```

Comment:
This will cause a massive amount of warnings.

I would proceed as following instead:
- introduce the `as` prop without touching the other props
- wait for a new major bump
- deprecate the `elementType` prop and add the warning
- wait for a new major bump
- remove the `elementType` prop

Ideally, we would also provide a codemod to run the migration.

Reply:
I think this is what was aimed for here and also agreed upon before. Generally, I see the intend here to warn through prop-types given the prop is used. Then fix up the warnings (while still supporting the prop) by migrating to `as` and then in the next major dropping the `elementType`.

How did you read us wanting a quick breaking change from this code?

Reply:
I think providing a codemod is a bit overkill considering we could make the changes in the MC-FE in about 2 minutes with find and replace. 

I think the path described by @tdeekens seems quite reasonable.

Reply:
Yeah, would be a nice learning experience to write a codemod during tech time if anybody wants but seems faster without. Assuming that not too many people use ui-kit yet (whom the find and replace is also quick an easy).

I am just lagging behind to how the suggested implementation here seemed to suggest another path than outlined above.